src/components/Common/ImageUpload/useUploadImages.jsx

```diff
@@ -0,0 +1,199 @@
+            });
+        } catch (error) {
+            console.log(`Error uploading this image:${error}`);
+        }
+    };
+
+    const handleRemove = (index) => {
+        setImages((prevImages) => {
+            const updatedImages = [...prevImages];
+            updatedImages.splice(index, 1);
+            return updatedImages;
+        });
+        setImagesName((prevImagesName) => {
+            const updatedImagesName = [...prevImagesName];
+            updatedImagesName.splice(index, 1);
+            return updatedImagesName;
+        });
+    };
+
+    const handleMultipleSingularOptions = async (imageFiles, multiple, maxImages) => {
+        if (multiple) return await handleManyUploads(imageFiles, maxImages);
+        const { image, imageName } = await handleUploadImage(imageFiles[0]);
+        setImages([image]);
+        setImagesName([imageName]);
+    };
+
+    const handleChange = async (e, multiple = false, maxImages = undefined) => {
+        const imageFiles = e.target.files;
+        await handleMultipleSingularOptions(imageFiles, multiple, maxImages);
+    };
+
+    const handleDrop = async (e, multiple = false, maxImages = undefined) => {
+        e.preventDefault();
+        const imageFiles = e.dataTransfer.files;
+        await handleMultipleSingularOptions(imageFiles, multiple, maxImages);
+    };
+
+    const handlePaste = async (e, multiple = false, maxImages = undefined) => {
+        const items = (e.clipboardData || e.originalEvent.clipboardData).items;
+        const imageFiles = [];
+
+        // Check if the paste event contains an image
+        for (const item of items) {
+            if (item.type.startsWith("image")) {
+                const file = item.getAsFile();
+                return imageFiles.push[file];
+            }
+        }
+
+        await handleMultipleSingularOptions(imageFiles, multiple, maxImages);
+    };
+
+    const handleDragOver = (e) => {
+        e.preventDefault();
+    };
+
+    const handleSubmit = async (file) => {
+        try {
+            const formData = new FormData();
+            formData.append("image", file);
+            const API_URL = getApiUrl("api/upload");
+            await axios.post(API_URL, formData);
+        } catch (err) {
+            if (err.message === "Request failed with status code 429") {
+                toast.error("You are uploading images too fast. Please wait a few seconds and try again.");
+            }
+        }
+    };
+
+    const handleSubmitManyImages = async (files) => {
+        for (const file of files) {
+            await handleSubmit(file);
+        }
+    };
+
+    const handleSubmitFromContent = async (content) => {
+        const imageTags = content.match(/<img[^>]*src="([^"]*)"[^>]*>/g) || [];
+        let updatedContent = content;
+        if (imageTags.length > 0) {
+            async function uploadContentImages() {
+                const imageUrls = imageTags.map((tag) => {
```

Comment:
A bit confused about why you have `uploadContentImages` declared inside this function and also outside